src/Aspire.Hosting.Milvus/MilvusServerResource.cs

```diff
@@ -0,0 +1,43 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire.Hosting.ApplicationModel;
+
+namespace Aspire.Hosting.Milvus;
+
+/// <summary>
+/// A resource that represents a Milvus database.
+/// </summary>
+public class MilvusServerResource : ContainerResource, IResourceWithConnectionString
+{
+    /// <summary>
+    /// Initializes a new instance of the <see cref="MilvusServerResource"/> class.
+    /// </summary>
+    /// <param name="name">The name of the resource.</param>
+    /// <param name="apiKey">A <see cref="ParameterResource"/> that contains the authentication apiKey/token</param>
+    public MilvusServerResource(string name, ParameterResource apiKey) : base(name)
+    {
+        ArgumentNullException.ThrowIfNull(apiKey);
+        ApiKeyParameter = apiKey;
+    }
+
+    internal const string PrimaryEndpointName = "grpc";
```

Comment:
(nit) for consistency, it is customary to put these const strings at the top of the class.